runtime/Cargo.toml

```diff
@@ -80,6 +80,7 @@ deno_web.workspace = true
 deno_webidl.workspace = true
 deno_websocket.workspace = true
 deno_webstorage.workspace = true
+rustyline = { version = "=10.0.0", default-features = false, features = ["custom-bindings"] }
```

Comment:
Can you explain why this dependency is needed?

Reply:
> Can you explain why this dependency is needed?

The current text input used for `prompt`, based on `stdin.readSync`, has a lot of bugs that add up to poor UX. For example:
1. Pressing <kbd>←</kbd> inputs an ANSI escape rather than navigating to the previous character. Same goes for other arrow keys, <kbd>Home</kbd>,  <kbd>End</kbd>, etc.
2. All non-ASCII characters are somehow treated as being 2 characters when pressing backspace. The first backspace corrupts the character to U+FFFD replacement character, then the second backspace deletes that. Further, the cursor still moves back by 1 column after each backspace keypress, so what's visually displayed is inconsistent with the underlying value, and you can even backspace into the prompt text itself.
3. Backspacing moves the cursor back by 1 column even for double-physical-width characters (CJK, emojis, etc). This partially "fixes" the second issue for these characters by moving the cursor inside the character when it's corrupted to U+FFFD, which is arguably a better representation of what happened.
4. Zero-width characters such as ZWNJ take up 1 column when they should take up 0.

Possibly some of these should be opened as separate bug issues, but presumably at least the first one is by-design.

`rustyline` handles all those details (and probably a load more I haven't thought of) correctly. While it'd be possible to re-implement all of that in JS using stdin raw mode, it would add significant complexity and mean shipping a lot of extra JavaScript as part of the runtime. `rustyline` is already a dependency of the repl, so I don't think it should increase binary sizes (haven't tested that yet though)... it's possible it'd increase binary sizes of bundled executables, though, if those don't include the repl by default?